api/objects.proto

```diff
@@ -267,3 +268,29 @@ message Secret {
 	// Whether the secret is an internal secret (not set by a user) or not.
 	bool internal = 4;
 }
+
+message IndexEntry {
+	string key = 1;
+	string val = 2;
+}
+
+// Generic is a top-level object with externally defined content and indexing.
+// SwarmKit can serve as a store for these objects without understanding their
+// meanings.
+message Generic {
+	string id = 1 [(gogoproto.customname) = "ID"];
+
+	Meta meta = 2 [(gogoproto.nullable) = false];
+
+	// Type is a string that identifies the type of data encoded in the
+	// payload.
+	string type = 3;
```

Comment:
This has a slightly different intention than the type info encoded in `Any`. This is basically the table name in our database. You could have multiple tables that have the same underlying data type. I should probably rename this and clarify the comment.